fbpic/openpmd_diag/boosted_particle_diag.py

```diff
@@ -190,18 +191,66 @@ def flush_to_disk(self):
             # over time into a single array
             for species_name in self.species_dict:
 
-                if snapshot.buffered_slices[species_name] != []:
```

Comment:
Note: I had to get rid of the above `if` condition. Otherwise, if one proc does not have any buffered slices for this snapshot (but other procs do), then this proc will not participate in the MPI gather below, and the other procs will just for the mpi gather indefinitely.